src/net.cpp

```diff
@@ -204,7 +204,8 @@ void AdvertiseLocal(CNode *pnode)
 // learn a new local address
 bool AddLocal(const CService& addr, int nScore)
 {
-    if (!addr.IsRoutable())
+    // TODO: skip IsRoutable only in regtest mode
+    if (!addr.IsRoutable() && nScore < LOCAL_MANUAL)
```

Comment:
Should probably use `Params().RequireRoutableExternalIP()` here as well if this is still needed for tests to pass.

Reply:
Surely! I forgot to commit `net.cpp`... Thanks.

Reply:
Hmm.. shouldn't `nScore < LOCAL_MANUAL` go away here then? Also, `// TODO:` above is not a `TODO` anymore I guess ;)

Reply:
Yes, I will remove `TODO` when tests finished. Actually, with `nScore < LOCAL_MANUAL` logic is unchanged even in regtest mode (for other tests without `-expernalip` parameter). But I think it can be removed safely...

Reply:
Hmm... Untouched tests failed. Seems like port 11211 is busy for some reason. Same problem with this port [was noticed](https://travis-ci.org/dashpay/dash/jobs/273261169) yesterday.